### PR TITLE
DEV-812 Make naming of exceptions more standard

### PR DESCRIPTION
**Pull request by ZHANGChunwang**

### CODE REVIEW

File: adapter-deterlab/src/main/java/sg/ncl/adapter/deterlab/AdapterDeterLab.java [hunk 1]
```diff
@@ -58,9 +60,12 @@ public String joinProjectNewUsers(String jsonString) {
         ResponseEntity response;
         try {
             response = restTemplate.exchange(properties.getJoinProjectNewUsers(), HttpMethod.POST, request, String.class);
-        } catch (RestClientException e) {
-            log.warn("DeterLab connection error new user join project: {}", e);
-            throw new AdapterDeterlabConnectException();
+        } catch (ResourceAccessException rae) {
```
Comment (ChrisZhong) [inline adapter-deterlab/src/main/java/sg/ncl/adapter/deterlab/AdapterDeterLab.java]: MAJOR Either log or rethrow this exception. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)

File: adapter-deterlab/src/main/java/sg/ncl/adapter/deterlab/exceptions/AdapterDeterLabConnectionFailedException.java [hunk 2]
```diff
@@ -2,19 +2,19 @@
...
 import org.springframework.web.bind.annotation.ResponseStatus;
-import sg.ncl.common.exception.base.NotFoundException;
+import org.springframework.web.client.ResourceAccessException;
 
 /**
  * @author Te Ye
  */
 @ResponseStatus(value = HttpStatus.REQUEST_TIMEOUT, reason = "Connection refused to remote adapter deterlab")
-public class AdapterDeterlabConnectException extends NotFoundException {
+public class AdapterDeterLabConnectionFailedException extends ResourceAccessException {
```
Comment (ChrisZhong) [inline adapter-deterlab/src/main/java/sg/ncl/adapter/deterlab/exceptions/AdapterDeterLabConnectionFailedException.java]: MAJOR This class has 7 parents which is greater than 5 authorized. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AMaximumInheritanceDepth)

File: service-registration/src/main/java/sg/ncl/service/registration/exceptions/TeamIdNullOrEmptyException.java [hunk 3]
```diff
@@ -2,10 +2,15 @@
 
 import org.springframework.http.HttpStatus;
 import org.springframework.web.bind.annotation.ResponseStatus;
+import sg.ncl.common.exception.base.BadRequestException;
 import sg.ncl.common.exception.base.NotFoundException;
```
Comment (ChrisZhong) [inline service-registration/src/main/java/sg/ncl/service/registration/exceptions/TeamIdNullOrEmptyException.java]: MINOR Remove this unused import 'sg.ncl.common.exception.base.NotFoundException'. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: service-registration/src/main/java/sg/ncl/service/registration/exceptions/IncompleteRegistrationFormException.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
+package sg.ncl.service.registration.exceptions;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import sg.ncl.common.exception.base.BadRequestException;
+import sg.ncl.common.exception.base.NotFoundException;
```
Comment (ChrisZhong) [inline service-registration/src/main/java/sg/ncl/service/registration/exceptions/IncompleteRegistrationFormException.java]: MINOR Remove this unused import 'sg.ncl.common.exception.base.NotFoundException'. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by ChrisZhong** [comment]
SonarQube analysis reported 5 issues
* MAJOR 2 major
* MINOR 3 minor

Watch the comments in this conversation to review them.

File: service-experiment/src/main/java/sg/ncl/service/experiment/exceptions/ExperimentNotFoundException.java [hunk 5]
```diff
@@ -1,11 +1,12 @@
...
-import org.springframework.http.HttpStatus;
-import org.springframework.web.bind.annotation.ResponseStatus;
 import sg.ncl.common.exception.base.NotFoundException;
 
 /**
  * Created by Desmond.
  */
-@ResponseStatus(code = HttpStatus.NOT_FOUND, reason = "There are no experiments.")
-public class ExperimentNotFoundException extends NotFoundException {}
+public class ExperimentNotFoundException extends NotFoundException {
```
Comment (teye) [inline service-experiment/src/main/java/sg/ncl/service/experiment/exceptions/ExperimentNotFoundException.java]: add response status

File: adapter-deterlab/src/main/java/sg/ncl/adapter/deterlab/exceptions/ExperimentStartException.java [hunk 6]
```diff
@@ -5,9 +5,9 @@
 /**
  * @author Te Ye
  */
-public class ExpStartException extends BadRequestException {
+public class ExperimentStartException extends BadRequestException {
```
Comment (teye) [inline adapter-deterlab/src/main/java/sg/ncl/adapter/deterlab/exceptions/ExperimentStartException.java]: add response status

File: adapter-deterlab/src/main/java/sg/ncl/adapter/deterlab/exceptions/ExperimentDeleteException.java [hunk 7]
```diff
@@ -5,9 +5,9 @@
 /**
  * @author Te Ye
  */
-public class ExpDeleteException extends BadRequestException {
+public class ExperimentDeleteException extends BadRequestException {
```
Comment (teye) [inline adapter-deterlab/src/main/java/sg/ncl/adapter/deterlab/exceptions/ExperimentDeleteException.java]: add response status

File: service-user/src/main/java/sg/ncl/service/user/exceptions/VerificationEmailNotMatchException.java [hunk 8]
```diff
@@ -5,9 +5,9 @@
 /**
  * Created by dcszwang on 8/12/2016.
  */
-public class EmailNotMatchException extends BadRequestException {
+public class VerificationEmailNotMatchException extends BadRequestException {
```
Comment (teye) [inline service-user/src/main/java/sg/ncl/service/user/exceptions/VerificationEmailNotMatchException.java]: add response status

File: service-registration/src/main/java/sg/ncl/service/registration/exceptions/InvalidTeamStatusException.java [hunk 9]
```diff
@@ -9,4 +9,7 @@
  */
 @ResponseStatus(code = HttpStatus.BAD_REQUEST, reason = "Team Status Invalid")
```
Comment (teye) [inline service-registration/src/main/java/sg/ncl/service/registration/exceptions/InvalidTeamStatusException.java]: change reason to Team status is invalid, make it consistent with the error message

**Pull request merged**